app/controllers/administrator/products_controller.rb

```diff
@@ -0,0 +1,49 @@
+class Administrator::ProductsController < Administrator::BaseController
+  before_action :set_product, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @products = Product.all
+  end
+
+  def show
```

Comment:
if you have actions that have no code blocks you can save some space by writing `def show; end`

Reply:
solved. ✓.
checked the others too ✓.